src/domain.ts

```diff
@@ -69,19 +59,39 @@ export default class DomainClient {
     );
   }
 
+  private resolveDomainType(domainType?: DomainType | string): DomainType | undefined {
+    if (domainType == null) {
+      return undefined;
+    }
+    
+    if (typeof domainType === 'string') {
+      const resolved = DomainType[domainType as keyof typeof DomainType];
+      if (resolved === undefined) {
+        throw new Error(`Invalid domain type: ${domainType}. Expected ALLOWED_EMAIL_DOMAIN or ORGANIZATION_DOMAIN`);
+      }
+      return resolved;
+    }
+    return domainType;
+  }
```

Comment:
lets return early for non-string case
```suggestion
private resolveDomainType(
  domainType?: DomainType | string
): DomainType | undefined {
  if (domainType == null) return;
  if (typeof domainType !== 'string') return domainType;
  const resolved = DomainType[domainType as keyof typeof DomainType];
  if (resolved === undefined) {
    throw new Error(
      `Invalid domain type: ${domainType}. Expected ALLOWED_EMAIL_DOMAIN or ORGANIZATION_DOMAIN`
    );
  }

  return resolved;
}
```